File: iotprotocol/iot_pronte/Program/publisher.js
const mqtt = require('mqtt')
const client = mqtt.connect('mqtt://localhost:1883');

const TOPIC = "diot";

const OPTIONS = { qos : 0, retain: true};

client.on('connect',function () {
  console.log("Broker Connected");
  client.publish(TOPIC, '{temp:24.5 C }',OPTIONS , function(err){
    if(!err) {
      console.log("Published Sucess");
    }
  }
  );
});

client.on('disconnect',()=> {
  console.log("Publisher disconnected!!!!!");
  
});

File: iotprotocol/IotProtocol/Mqtt_prg/publisher.py
import paho.mqtt.client as mqtt
import time
print("starting publisher...")

topic = input("Enter the topic to publish:")
Qos=0
client=mqtt.Client()
def on_connect(client,userdata,flags,rc):
    time.sleep(6)
    if rc==0:
        print("connected to broker")
        client.publish(topic,"hello",Qos)

client.on_connect = on_connect(client,None,1,0)


client.connect("localhost",1883)
